Gordon360/Controllers/HousingController.cs

```diff
@@ -207,17 +207,16 @@ public ActionResult<ApartmentApplicationViewModel> GetApartmentApplication(int a
             //get token data from context, username is the username of current logged in person
             var authenticatedUserUsername = AuthUtils.GetUsername(User);
 
-            string username = _accountService.GetAccountByUsername(authenticatedUserUsername).ADUserName;
-
             bool isAdmin = false;
 
             try
             {
-                isAdmin = _administratorService.GetByUsername(username) != null;
+                isAdmin = _administratorService.GetByUsername(authenticatedUserUsername) != null
+                    || _housingService.CheckIfHousingAdmin(authenticatedUserUsername);
             }
             catch
             {
-                isAdmin = _housingService.CheckIfHousingAdmin(username);
+                isAdmin = false;
```

Comment:
I removed the error throw from administratorService.GetByUsername() for when the username is not found. Now it just returns null. So technically the try catch is now just there to make sure that we set isAdmin. It should probably just be removed and we can return an error to the UI, eh?

Reply:
Yeah, I see now that you removed the error being thrown from the AdminService. In that case, I do think that we remove the try/catch entirely - because now it is only serving to swallow up more insidious errors. Rather than mutating `isAdmin`, you can just set it directly.
```csharp
bool isAdmin = _administratorService.GetByUsername(authenticatedUserUsername) != null || _housingService.CheckIfHousingAdmin(authenticatedUserUsername);
```

Reply:
Wouldn't it be 
```csharp
bool isAdmin = _administratorService.GetByUsername(authenticatedUserUsername) || _housingService.CheckIfHousingAdmin(authenticatedUserUsername);
```
(removing `!= null`)

Reply:
Done.